sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveGroupByAll.scala

```diff
@@ -93,8 +93,9 @@ object ResolveGroupByAll extends Rule[LogicalPlan] {
    * end of analysis, so we can tell users that we fail to infer the grouping columns.
    */
   def checkAnalysis(operator: LogicalPlan): Unit = operator match {
-    case a: Aggregate if matchToken(a) =>
-      if (a.aggregateExpressions.exists(_.exists(_.isInstanceOf[Attribute]))) {
+    case a: Aggregate if a.aggregateExpressions.forall(_.resolved) && matchToken(a) =>
```

Comment:
Nit: there are some codes duplicated with the `apply` method. We can consider simpler refactors.

Reply:
The check code here is a partial version of the `apply` code. I can't think of a simple refactor that can reduce more code duplication. Let's think about this later.

Reply:
You are right about the check here. For the grouping expression inference, I think we can refactor it as https://github.com/apache/spark/pull/39523